tensorflow_probability/python/internal/backend/numpy/numpy_math.py

```diff
@@ -823,7 +823,7 @@ def _multiply_no_nan(x, y, name=None):  # pylint: disable=unused-argument
 def _softplus(x, name=None):  # pylint: disable=unused-argument
   # TODO(b/146563881): Investigate improving numerical accuracy here.
   if not JAX_MODE:
-    return np.log1p(np.exp(-np.abs(x))) + np.maximum(x, 0.)
+    return np.log1p(np.exp(-np.abs(x))) + np.maximum(x, np.zeros_like(x))
```

Comment:
It was done to have tests passed on numpy backend (see #861 ), since it's ignored anyway (notap) - reverted.

Reply:
Understood, thanks. At the moment our focus is on the JAX backend. Fixing dtypes in the numpy backend will require some rewrite over a lot of our code (particularly because numpy is more loose with dtypes), and so we might do that as a separate clean up.